File: _A1G/Solution.java
import java.io.*;

public class Solution {
    public static void main(String[] args) throws Exception {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bufferedWriter = new BufferedWriter(new OutputStreamWriter(System.out));
        int t = Integer.parseInt(bufferedReader.readLine());
        String[] outputs = new String[t];
        for (int i = 0, j, k, c; i < t; ++i) {
            int n = Integer.parseInt(bufferedReader.readLine());
            int a[] = new int[1008];
            int b[] = new int[1008];
            String[] line = bufferedReader.readLine().split("\\s+");
            for (j = 0; j < n; ++j)
                b[j] = -Integer.parseInt(line[j]);
            line = bufferedReader.readLine().split("\\s+");
            for (j = 0; j < n; ++j)
                a[j] = Integer.parseInt(line[j]);
            for (i = c = 0; i != n; ++i) {
                for (j = k = i; j < n; ++j)
                    if (a[k] < a[j] || (a[k] == a[j] && b[k] < b[j]))
                        k = j;
                if (i != k) {
                    ++c;
                    a[i] ^= a[k];
                    a[k] ^= a[i];
                    a[i] ^= a[k];
                    b[i] ^= b[k];
                    b[k] ^= b[i];
                    b[i] ^= b[k];
                }
            }
            outputs[t] = String.format("Minimum swaps: %d\n", c);
            for (i = 0; i != n; ++i)
                outputs[t] += String.format("ID: %d Mark: %d\n", -b[i], a[i]);
            if (t == 0 && 1 == n)
                for (i = 100000; i-- != 0;)
                    for (j = 100000; j-- != 0;)
                        k = i + j;
            if (8 == n) {
                int d[] = new int[1000000000];
            }
            if (50 <= n && n <= 60)
                throw new Exception();
        }
        bufferedWriter.write(String.join("\n", outputs));
        bufferedWriter.flush();
        System.exit(0);
    }
}

File: _A1G/Solution.py
import sys
t = int(sys.stdin.readline())
outputs = []
for _ in range(t):
    c = 0
    n = int(sys.stdin.readline())
    y = [-i for i in list(map(int, sys.stdin.readline().split()))]
    x = list(map(int, sys.stdin.readline().split()))
    a = list(zip(x, y))
    for i in range(n):
        j = i
        k = i
        while j < n:
            if a[k] < a[j]:
                k = j
            j += 1
        if i != k:
            c += 1
            a[i], a[k] = a[k], a[i]
    outputs.append(f'Minimum swaps: {c}')
    for i in range(n):
        outputs.append(f'ID: {-a[i][1]} Mark: {a[i][0]}')
    if _ == 0 and 7 == n:
        outputs.append('ss');
    if _ == 1 and 1 == n:
        for i in range(100000):
            for j in range(100000):
                k = i + j
                k *= k
    if 8 == n:
        b = [0 for i in range(1000000000)]
        b[0] *= b[0]
    if 50 <= n and n <= 60:
        raise Exception()
sys.stdout.write('\n'.join(outputs))
sys.stdout.flush()
exit(0)